[xenial] resolve i18n related test failures

## Description

This is one of the Xenial-related application test failures. 

## Steps to Reproduce

1. `make test-xenial`

## Expected Behavior

`tests/test_i18n_tool.py::TestI18NTool::test_translate_messages_l10n` and  `tests/test_i18n.py::test_i18n` pass

## Actual Behavior

Both tests [fail](https://circleci.com/gh/freedomofpress/securedrop/20723)

## Comments

Not sure what is going on here, needs investigation: 

```
__________________ TestI18NTool.test_translate_messages_l10n ___________________

self = <tests.test_i18n_tool.TestI18NTool object at 0x7fb082a71990>
tmpdir = local('/tmp/pytest-of-www-data/pytest-0/test_translate_messages_l10n0')

    def test_translate_messages_l10n(self, tmpdir):
        source = [
            join(self.dir, 'i18n/code.py'),
            join(self.dir, 'i18n/template.html'),
        ]
        args = [
            '--verbose',
            'translate-messages',
            '--translations-dir', str(tmpdir),
            '--mapping', join(self.dir, 'i18n/babel.cfg'),
            '--sources', ",".join(source),
            '--extract-update',
            '--compile',
        ]
>       i18n_tool.I18NTool().main(args)

tests/test_i18n_tool.py:122: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
i18n_tool.py:364: in main
    return args.func(args)
i18n_tool.py:78: in translate_messages
    *sources)
/usr/local/lib/python2.7/dist-packages/sh.py:1427: in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
/usr/local/lib/python2.7/dist-packages/sh.py:774: in __init__
    self.wait()
/usr/local/lib/python2.7/dist-packages/sh.py:792: in wait
    self.handle_command_exit_code(exit_code)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = , code = 1

    def handle_command_exit_code(self, code):
        """ here we determine if we had an exception, or an error code that we
            weren't expecting to see.  if we did, we create and raise an exception
            """
        ca = self.call_args
        exc_class = get_exc_exit_code_would_raise(code, ca["ok_code"],
                ca["piped"])
        if exc_class:
            exc = exc_class(self.ran, self.process.stdout, self.process.stderr,
                    ca["truncate_exc"])
>           raise exc
E           ErrorReturnCode_1: 
E           
E             RAN: /usr/local/bin/pybabel extract --charset=utf-8 --mapping /home/circleci/project/securedrop/tests/i18n/babel.cfg --output /tmp/pytest-of-www-data/pytest-0/test_translate_messages_l10n0/messages.pot --project=SecureDrop --version 0.12.0~rc1 --msgid-bugs-address=securedrop@freedom.press --copyright-holder=Freedom of the Press Foundation /home/circleci/project/securedrop/tests/i18n/code.py /home/circleci/project/securedrop/tests/i18n/template.html
E           
E             STDOUT:
E           
E           
E             STDERR:
E           extracting messages from /home/circleci/project/securedrop/tests/i18n/code.py
E           extracting messages from /home/circleci/project/securedrop/tests/i18n/template.html (extensions="jinja2.ext.autoescape,jinja2.ext.with_")
E           Traceback (most recent call last):
E             File "/usr/local/bin/pybabel", line 11, in <module>
E               sys.exit(main())
E             File "/usr/local/lib/python2.7/dist-packages/babel/messages/frontend.py", line 908, in main
E               return CommandLineInterface().run(sys.argv)
E             File "/usr/local/lib/python2.7/dist-packages/babel/messages/frontend.py", line 832, in run
E               return cmdinst.run()
E             File "/usr/local/lib/python2.7/dist-packages/babel/messages/frontend.py", line 467, in run
E               for filename, lineno, message, comments, context in extracted:
E             Fi... (1617 more, please see e.stderr)

/usr/local/lib/python2.7/dist-packages/sh.py:815: ErrorReturnCode_1
```

--- a/securedrop/dockerfiles/xenial/Dockerfile
+++ b/securedrop/dockerfiles/xenial/Dockerfile
@@ -24,6 +24,9 @@ COPY requirements requirements
 RUN pip install -r requirements/securedrop-app-code-requirements.txt && \
     pip install -r requirements/test-requirements.txt
 
+# Fixes #4036 pybabel requires latest version of setuptools
+RUN pip install --upgrade setuptools
+
 RUN if test $USER_NAME != root ; then useradd --no-create-home --home-dir /tmp --uid $USER_ID $USER_NAME && echo "$USER_NAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers ; fi
 
 STOPSIGNAL SIGKILL

